source/solvers/fluid_dynamics_matrix_free_operators.cc

```diff
@@ -830,6 +1044,130 @@ NavierStokesOperatorBase<dim, number>::do_cell_integral_range(
     }
 }
 
+template <int dim, typename number>
+void
+NavierStokesOperatorBase<dim, number>::do_internal_face_integral_range(
+  const MatrixFree<dim, number>               &matrix_free,
+  VectorType                                  &dst,
+  const VectorType                            &src,
+  const std::pair<unsigned int, unsigned int> &range) const
+{
+  (void)matrix_free;
+  (void)dst;
+  (void)src;
+  (void)range;
+
+  // nothing to do
+}
+
+template <int dim, typename number>
+template <bool assemble_residual>
+void
+NavierStokesOperatorBase<dim, number>::do_weak_dirichlet_boundary_range(
+  const MatrixFree<dim, number>               &matrix_free,
+  VectorType                                  &dst,
+  const VectorType                            &src,
+  const std::pair<unsigned int, unsigned int> &range) const
+{
+  FEFaceIntegrator phi(matrix_free, true, 0);
+
+  for (auto cell = range.first; cell < range.second; ++cell)
+    {
+      phi.reinit(cell);
+
+      if (assemble_residual)
+        phi.read_dof_values_plain(src);
+      else
+        phi.read_dof_values(src);
+
+      do_local_weak_dirichlet_bc<assemble_residual>(phi);
+
+      phi.distribute_local_to_global(dst);
+    }
+}
+
+
+// Add variant for face terms residual
```

Comment:
Could you add here the equations being calculated here? I know they are in the comments below, but I find it useful to have them here as well. Then it would be uniform and similar to the functions in the `NavierStokesStabilizedOperator`

Reply:
Sure thing :)